test/unit/test_inference.py

```diff
@@ -399,3 +399,15 @@ def test_topk_func(batch_size, beam_size, target_vocab_size):
     assert all(mx_hyp == np_hyp)
     assert all(mx_word == np_word)
     assert all(mx_values == np_values)
+
+
+def test_get_best_word_indeces_for_kth_hypotheses():
+    np.random.seed(13)
```

Comment:
how about running the test on the minimal example you had in order to test for the actual outputs?

Reply:
yeah, probably makes sense. This way we can at least assert on the correctness of the outputs. The way the test is right now its not really helpful :)